gateserver/controller_api.py

```diff
@@ -0,0 +1,41 @@
+from .controller_protocol import *
+from . import db
+from . import utils
+
+def get_key_for_mac(mac):
+    """Loads the key for this controller from the DB."""
+    rs = db.exec_sql('SELECT key FROM controller WHERE id = %s',
+                     (utils.bytes2mac(mac),), ret=True)
+    checkmsg(len(rs) == 1, 'unknown controllerID')
+    return rs[0]['key'].tobytes()
+
+def fstring(buf):
+    """See https://github.com/fmfi-svt-gate/server/wiki/Controller-%E2%86%94-Server-Protocol#note-isic-ids-representation"""
+    length, string = int(buf[0]), buf[1:]
+    checkmsg(length <= len(string), 'fstring length > buffer size')
+    return string[:length]
+
+process_request = {
+    MsgType.OPEN:
+        lambda data: ((S.OK if fstring(data) == b'Hello' else S.ERR), None),
```

Comment:
Hmm. Zaujima ma, ako bude vyzerat to "ozajstne". Aj ked je samotne telo len docasny priklad, daj mu "ozajstnu" formu.
(Lebo potom sa mozme bavit o dalsich veciach, napr ci argumenty handlovacej fcie budu len data alebo cely Request, a ci vysledok bude (status,data) alebo cely Response, apod. Teda ak pouzijes ten napad nizsie ze aj tail bude v structe.)